.github/workflows/nia-e2e.yaml

```diff
@@ -0,0 +1,78 @@
+name: Now in Android CI
+
+on:
+  workflow_call:
+    inputs:
+      android-version:
+        default: "13"
+        description: Set android version to use
+        required: false
+        type: string
+      use-vm:
+        default: "false"
+        description: Run Android in a VM instead of a system container
+        required: false
+        type: string
+
+
+permissions:
+  contents: read
+
+jobs:
+  nia-e2e:
+    name: Setup Anbox and Run E2E Tests
+    runs-on: ubuntu-latest
+    env:
+        ANDROID_VERSION: ${{ inputs.android-version }}
+        USE_VM: ${{ inputs.use-vm }}
+    timeout-minutes: 60
+    defaults:
+      run:
+        working-directory: ./nowinandroid
+    
+    steps:
+    - name: Checkout
+      uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+    - name: Setup Anbox Cloud
+      uses: canonical/anbox-cloud-github-action@dc99b94280cfcf44a86b62aeac25ba34f14725e0
+      with:
+        channel: latest/stable
+
+    - name: Tune Anbox Cloud
+      run: |
+        amc config set container.security_updates false
+
+    - name: Restore cached images
+      uses: actions/cache/restore@5a3ec84eff668545956fd18022155c47e93e2684 # v4
+      with:
+        path: images
+        key: anbox-images-amd64
+
+    - name: Import cached images
+      run: |
+        set -x
+        amc image add jammy:android${ANDROID_VERSION}:amd64 ./images/android${ANDROID_VERSION}.tar.xz
+
+    - name: Create android instance
+      id: create-instance
+      run: |
+        set -x
+        id="$(amc launch -r -s adb jammy:android${ANDROID_VERSION}:amd64)"
+        amc wait -c status=running "$id"
+        echo "id=$id" >> "$GITHUB_OUTPUT"
+
+    - name: Access android over adb
+      run: |
+        sudo apt install -y adb
+        id=${{ steps.create-instance.outputs.id }}
+        addr="$(amc show "$id" --format=json | jq -r .network.address)"
+        adb connect "$addr":5559
+        test "$(adb shell getprop ro.product.model)" = Anbox
+
+    - name: Run e2e tests
+      run: ./gradlew :app:installDemoDebug
```

Comment:
This only installs the APK but doesn't run any tests, we need `connectedDemoDebugAndroidTest` for that see https://github.com/android/nowinandroid?tab=readme-ov-file#testing